### PR TITLE
Fix compiler errors in tests and clippy warnings

### PR DESCRIPTION
**Pull request by kettlebell**
Got it down to 11 warnings.

### CODE REVIEW

File: spectrum-network/tests/peer_manager/peer_state_tests.rs [hunk 1]
```diff
@@ -1,51 +1,61 @@
...
 
 #[test]
 fn should_connect_to_peer_when_vacant_connections_available() {
     let mut peer_state = mk_peers_state(4, 2, 10);
-    let peer_id = PeerId::random();
+    let peer_id = PeerDestination::PeerId(PeerId::random());
 
-    let peer = peer_state.try_add_peer(peer_id, false);
-    let connected_peer = peer.unwrap().connect();
-    assert_eq!(connected_peer.is_ok(), true);
```
Comment (kettlebell) [inline spectrum-network/tests/peer_manager/peer_state_tests.rs:33]: `connect()` is now infallible?

File: spectrum-network/tests/peer_manager/peer_state_tests.rs [hunk 2]
```diff
@@ -1,51 +1,61 @@
...
 #[test]
 fn err_connect_to_peer_when_vacant_connections_not_available() {
     let mut peer_state = mk_peers_state(0, 0, 10);
-    let peer_id = PeerId::random();
+    let peer_id = PeerDestination::PeerId(PeerId::random());
 
-    let peer = peer_state.try_add_peer(peer_id, false).unwrap();
-    assert!(peer.connect().is_err());
+    let peer = peer_state.try_add_peer(peer_id, false, false).unwrap();
+    let _connected_peer = peer.connect();
```
Comment (kettlebell) [inline spectrum-network/tests/peer_manager/peer_state_tests.rs:44]: And here too.

File: spectrum-network/tests/peer_manager/peer_state_tests.rs [hunk 3]
```diff
@@ -1,51 +1,61 @@
...
+    //assert!(peer.connect().is_err());
 }
 
-fn mk_peers_state(max_incoming: usize, max_outgoing: usize, capacity: usize) -> impl PeersState {
-    let pset_config = PeerIndexConfig {
-        max_incoming,
-        max_outgoing,
+fn mk_peers_state(max_inbound: usize, max_outbound: usize, capacity: usize) -> impl PeersState {
+    let netw_conf = NetworkingConfig {
+        min_known_peers: 2,
```
Comment (kettlebell) [inline spectrum-network/tests/peer_manager/peer_state_tests.rs:50]: I just made up these parameter values.

File: spectrum-network/tests/peer_manager/peer_state_tests.rs (hunk 1 above)
Comment (oskin1) [reply spectrum-network/tests/peer_manager/peer_state_tests.rs:33]: Yes, now PM calls `connect()` only when it 100% sure it's needed.

File: spectrum-network/tests/peer_manager/peer_state_tests.rs (hunk 3 above)
Comment (oskin1) [reply spectrum-network/tests/peer_manager/peer_state_tests.rs:50]: Totally ok in tests

File: spectrum-network/src/peer_manager/data.rs [hunk 4]
```diff
@@ -37,17 +37,17 @@ impl Serialize for PeerDestination {
 }
 
 impl<'de> Deserialize<'de> for PeerDestination {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    fn deserialize<D>(_deserializer: D) -> Result<Self, D::Error>
     where
         D: Deserializer<'de>,
     {
         todo!()
     }
```
Comment (oskin1) [inline spectrum-network/src/peer_manager/data.rs]: Please make sure to sync with the original branch, this Deserialize is implemented there already so no need for `_deserializer`

**Review by oskin1 — APPROVED** [review]

File: spectrum-network/src/peer_manager/data.rs (hunk 4 above)
Comment (kettlebell) [reply spectrum-network/src/peer_manager/data.rs]: Rebased.

**Pull request merged**